src/shared/internal-calendar/fixed-range/fixed-range-cell.tsx

```diff
@@ -0,0 +1,183 @@
+    hoverDate: string;
+    minDate?: string | undefined;
+    maxDate?: string | undefined;
+    disabledDates?: string[] | undefined;
+    allowDisabledSelection?: boolean | undefined;
+    numberOfDays?: number;
+    onSelect: (value: Dayjs, disabled: boolean) => void;
+    onHover: (value: string, disabled: boolean) => void;
+}
+
+export const FixedRangeDayCell = ({
+    date,
+    calendarDate,
+    selectedDate,
+    hoverDate,
+    minDate,
+    maxDate,
+    disabledDates,
+    allowDisabledSelection,
+    numberOfDays,
+    onSelect,
+    onHover,
+}: Props) => {
+    // =========================================================================
+    // CONSTS
+    // =========================================================================
+    const disabled = CalendarHelper.isDisabledDay(
+        date,
+        disabledDates,
+        minDate,
+        maxDate
+    );
+    const interactive = !disabled || allowDisabledSelection;
+    const { start: rangeStart, end: rangeEnd } = selectedDate
+        ? CalendarHelper.getFixedRangeStartEnd(
+              DateHelper.toDayjs(selectedDate),
+              numberOfDays
+          )
+        : { start: undefined, end: undefined };
+    const { start: hoverStart, end: hoverEnd } = hoverDate
+        ? CalendarHelper.getFixedRangeStartEnd(
+              DateHelper.toDayjs(hoverDate),
+              numberOfDays
+          )
+        : { start: undefined, end: undefined };
+
+    const isSelected =
+        selectedDate && date.isBetween(rangeStart, rangeEnd, "day", "[]");
+    const isHover =
+        hoverDate && date.isBetween(hoverStart, hoverEnd, "day", "[]");
+    const isStart =
+        (isSelected && date.isSame(rangeStart)) ||
+        (isHover && date.isSame(hoverStart));
+    const isEnd =
+        (isSelected && date.isSame(rangeEnd)) ||
+        (isHover && date.isSame(hoverEnd));
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleSelect = () => {
+        onSelect(date, !interactive);
+    };
+
+    const handleHover = () => {
+        onHover(date.format("YYYY-MM-DD"), !interactive);
+    };
+
+    // =========================================================================
+    // HELPERS
+    // =========================================================================
+    const getRangeStyle = (): CellStyleProps => {
+        const props: CellStyleProps = {};
+
+        let type: CellType = undefined;
+        if (isSelected && isHover) {
+            type = "overlap";
+        } else if (isSelected) {
+            type = "selected";
+        } else if (isHover) {
+            type = "hover-dash";
+        }
+
+        if (type) {
+            if (isStart && !isSelected) {
+                props.circleLeft = "hover-current";
+                props.circleRight = "hover-current";
+                props.circleShadow = true;
```

Comment:
hover style can be handled separately by checking date equals hoverDate. this would then cover the case of the box-shadow for the overlap case